airflow/contrib/hooks/gcs_hook.py

```diff
@@ -47,9 +45,11 @@ def get_conn(self):
         """
         Returns a Google Cloud Storage service object.
         """
-        http_authorized = self._authorize()
-        return build(
-            'storage', 'v1', http=http_authorized, cache_discovery=False)
+        if not self._conn:
+            self._conn = storage.Client(credentials=self._get_credentials(),
+                                        project=self.project_id)
```

Comment:
In other hooks, `project_id` is a method parameter. In this implementation, user can only pass `project_id` as a connection configuration. This introduces inconsistencies. What steps should we take to unify these situations for all GCP operator?

We have a 3 options:
1) Specifying `project_id` in connection configuration.
2) Specifying `project_id` in a method parameter with fallback to connection configuration
3) Specifying `project_id` in a hook constructor parameter with fallback to connection configuration.

The third variant does not appear anywhere, but it seems to me most expected. Initalizing parameters are not mixed with execution time parameters. `project_id` is a parameter that initialize client library. It don't execute a API call. 

Probably the wrong place for this discussion, but we should take steps to use each operator and hook for GCP to be identical. 

CC: @potiuk @antonimaciej

Reply:
Ya, let's discuss this and decide on this on the mailing list.

Reply:
Did we discuss this? `region_name` on various AWS hooks/operators have the same pattern (some take them as kwargs, some just from the connection)